### PR TITLE
chore: replace old deps checker in extension

### PR DESCRIPTION
**Pull request by meifans**
Replace old deps checker in extension.

### CODE REVIEW

File: packages/vscode-extension/src/debug/depsChecker/vscodeTelemetry.ts [hunk 1]
```diff
@@ -6,10 +6,9 @@ import { performance } from "perf_hooks";
 import { SystemError, UserError } from "@microsoft/teamsfx-api";
 import { TelemetryProperty } from "../../telemetry/extTelemetryEvents";
 import { ExtTelemetry } from "../../telemetry/extTelemetry";
-import { DepsCheckerEvent, TelemetryMessurement } from "./common";
-import { IDepsTelemetry } from "./checker";
+import { DepsTelemetry, DepsCheckerEvent, TelemetryMessurement } from "@microsoft/teamsfx-core";
 
-export class VSCodeTelemetry implements IDepsTelemetry {
+export class VscodeTelemetry implements DepsTelemetry {
```
Comment (kuojianlu) [inline packages/vscode-extension/src/debug/depsChecker/vscodeTelemetry.ts:11]: Better to keep naming consistency? I prefer `VSCodeTelemetry` which is consistent with `VSCodeDepsChecker` class in `vscodeChecker.ts`.

Comment (meifans) [reply packages/vscode-extension/src/debug/depsChecker/vscodeTelemetry.ts]: Good idea, changed.

File: packages/vscode-extension/src/debug/teamsfxTaskProvider.ts [hunk 2]
```diff
@@ -8,16 +8,18 @@ import * as constants from "./constants";
 import * as commonUtils from "./commonUtils";
 import { Json, ProductName, ProjectSettings, VsCodeEnv } from "@microsoft/teamsfx-api";
 import { LocalEnvManager } from "@microsoft/teamsfx-core";
-import { DotnetChecker } from "./depsChecker/dotnetChecker";
-import { FuncToolChecker } from "./depsChecker/funcToolChecker";
-import { NgrokChecker } from "./depsChecker/ngrokChecker";
+import { VSCodeDepsChecker } from "./depsChecker/vscodeChecker";
```
Comment (swatDong) [inline packages/vscode-extension/src/debug/teamsfxTaskProvider.ts:11]: Make the ts file names consistent, `vscodeXxx` or `VSCodeXxx`?

File: packages/vscode-extension/src/debug/depsChecker/vscodeHelper.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import { commands, MessageItem, Uri, window, workspace, WorkspaceConfiguration } from "vscode";
```
Comment (kuojianlu) [inline packages/vscode-extension/src/debug/depsChecker/vscodeHelper.ts:1]: Add file header.

Comment (swatDong) [inline packages/vscode-extension/src/debug/depsChecker/vscodeHelper.ts:1]: File header

File: packages/vscode-extension/src/debug/depsChecker/backendExtensionsInstall.ts [hunk 4]
```diff
@@ -1,89 +1,31 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-// NOTE:
```
Comment (swatDong) [inline packages/vscode-extension/src/debug/depsChecker/backendExtensionsInstall.ts:4]: Do we still need `tools/depsChecker/copyfiles`?

File: packages/vscode-extension/src/debug/depsChecker/vscodeHelper.ts (hunk 3 above)
Comment (meifans) [reply packages/vscode-extension/src/debug/depsChecker/vscodeHelper.ts:4]: Added

File: packages/vscode-extension/src/debug/depsChecker/backendExtensionsInstall.ts (hunk 4 above)
Comment (meifans) [reply packages/vscode-extension/src/debug/depsChecker/backendExtensionsInstall.ts:4]: deleted

**Comment by kuojianlu** [comment]
Do we still need `packages/vscode-extension/src/debug/depsChecker/resource/`?

**Review by swatDong — APPROVED** [review]

**Comment by meifans** [comment]
> Do we still need `packages/vscode-extension/src/debug/depsChecker/resource/`?

Good catch, deleted

**Comment by kuojianlu** [comment]
> > Do we still need `packages/vscode-extension/src/debug/depsChecker/resource/`?
> 
> Good catch, deleted

1. Delete `packages/vscode-extension/src/debug/depsChecker/resource`.
2. Update `packages/vscode-extension/webpack.config.js`.
3. Update `packages/vscode-extension/package.json` (copy-files script).

**Review by kuojianlu — APPROVED** [review]

**Review by HuihuiWu-Microsoft — APPROVED** [review]

**Pull request merged**